src/main/java/marquez/db/DatasourceDao.java

```diff
@@ -16,22 +16,26 @@
 
 import java.util.List;
 import java.util.Optional;
+import marquez.common.models.DatasourceName;
+import marquez.common.models.DatasourceUrn;
 import marquez.db.mappers.DatasourceRowMapper;
 import marquez.db.models.DatasourceRow;
 import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
 import org.jdbi.v3.sqlobject.customizer.BindBean;
 import org.jdbi.v3.sqlobject.statement.SqlQuery;
-import org.jdbi.v3.sqlobject.statement.SqlUpdate;
 
 @RegisterRowMapper(DatasourceRowMapper.class)
 public interface DatasourceDao {
-  @SqlUpdate(
+  @SqlQuery(
       "INSERT INTO datasources (guid, urn, name, connection_url) "
-          + "VALUES (:uuid, :urn, :name, :connectionUrl)")
-  void insert(@BindBean DatasourceRow datasourceRow);
+          + "VALUES (:uuid, :urn, :name, :connectionUrl) ON CONFLICT DO NOTHING RETURNING * ")
```

Comment:
Minor: I'd make the following changes to `@SqlQuery`:

1. Remove `ON CONFLICT DO NOTHING`. Since we now check to ensure a given datasource row already exists before insertion, we'll want to log occurrences where conflicts do happen rather than handling the conflict silently.
2. Move `RETURNING` to separate lines

```java
@SqlQuery(
    "INSERT INTO datasources (guid, urn, name, connection_url) "
        + "VALUES (:uuid, :urn, :name, :connectionUrl) "
        + "RETURNING * ")
```